File: sleepy-test/java-junit/TestInstrumentation_testTimer.java
public class TestInstrumentation extends TestCase {
    public void testTimer() throws Exception {
        Instrumentation.Timer timer = new Instrumentation.Timer();

        assertEquals(0, timer.getTicks());
        assertEquals(0, timer.getTotal());
        assertEquals(0, timer.getOwn());
        assertEquals(0, timer.getOwnAvg());
        assertEquals(0, timer.getTotalAvg());
        assertEquals(0, timer.getOwnSquareSum());
        assertEquals(0, timer.getTotalSquareSum());
        assertEquals(0, timer.getOwnMin());
        assertEquals(0, timer.getOwnMax());
        assertEquals(0, timer.getTotalMin());
        assertEquals(0, timer.getTotalMax());

        assertEquals(0, timer.getValue().getTicks());
        assertEquals(0, timer.getValue().getTotal());
        assertEquals(0, timer.getValue().getOwn());
        assertEquals(0, timer.getValue().getOwnAvg());
        assertEquals(0, timer.getValue().getTotalAvg());
        assertEquals(0, timer.getValue().getOwnSquareSum());
        assertEquals(0, timer.getValue().getTotalSquareSum());
        assertEquals(0, timer.getValue().getOwnMin());
        assertEquals(0, timer.getValue().getOwnMax());
        assertEquals(0, timer.getValue().getTotalMin());
        assertEquals(0, timer.getValue().getTotalMax());

        Instrumentation.Cron cron1 = new Instrumentation.Cron();
        cron1.start();
        Thread.sleep(INTERVAL);
        cron1.stop();
        timer.addCron(cron1);

        assertEquals(1, timer.getTicks());
        assertEquals(cron1.getTotal(), timer.getTotal());
        assertEquals(cron1.getOwn(), timer.getOwn());
        assertEquals(cron1.getOwn(), timer.getOwnAvg());
        assertEquals(cron1.getTotal(), timer.getTotalAvg());
        assertEquals(cron1.getOwn() * cron1.getOwn(), timer.getOwnSquareSum());
        assertEquals(cron1.getTotal() * cron1.getTotal(), timer.getTotalSquareSum());
        assertEquals(cron1.getOwn(), timer.getOwnMin());
        assertEquals(cron1.getOwn(), timer.getOwnMax());
        assertEquals(cron1.getTotal(), timer.getTotalMin());
        assertEquals(cron1.getTotal(), timer.getTotalMax());

        assertEquals(1, timer.getValue().getTicks());
        assertEquals(cron1.getTotal(), timer.getValue().getTotal());
        assertEquals(cron1.getOwn(), timer.getValue().getOwn());
        assertEquals(cron1.getOwn(), timer.getValue().getOwnAvg());
        assertEquals(cron1.getTotal(), timer.getValue().getTotalAvg());
        assertEquals(cron1.getOwn() * cron1.getOwn(), timer.getValue().getOwnSquareSum());
        assertEquals(cron1.getTotal() * cron1.getTotal(), timer.getValue().getTotalSquareSum());
        assertEquals(cron1.getOwn(), timer.getValue().getOwnMin());
        assertEquals(cron1.getOwn(), timer.getValue().getOwnMax());
        assertEquals(cron1.getTotal(), timer.getValue().getTotalMin());
        assertEquals(cron1.getTotal(), timer.getValue().getTotalMax());

        Instrumentation.Cron cron2 = new Instrumentation.Cron();
        cron2.start();
        Thread.sleep(INTERVAL * 2);
        cron2.stop();
        timer.addCron(cron2);

        assertEquals(2, timer.getTicks());
        assertEquals(cron1.getTotal() + cron2.getTotal(), timer.getTotal());
        assertEquals(cron1.getOwn() + cron2.getOwn(), timer.getOwn());
        assertEquals((cron1.getOwn() + cron2.getOwn()) / 2, timer.getOwnAvg());
        assertEquals((cron1.getTotal() + cron2.getTotal()) / 2, timer.getTotalAvg());
        assertEquals(cron1.getOwn() * cron1.getOwn() + cron2.getOwn() * cron2.getOwn(),
                     timer.getOwnSquareSum());
        assertEquals(cron1.getTotal() * cron1.getTotal() + cron2.getTotal() * cron2.getTotal(),
                     timer.getTotalSquareSum());
        assertEquals(cron1.getOwn(), timer.getOwnMin());
        assertEquals(cron2.getOwn(), timer.getOwnMax());
        assertEquals(cron1.getTotal(), timer.getTotalMin());
        assertEquals(cron2.getTotal(), timer.getTotalMax());
    }
}

File: sleepy-test/CSharpXUnit/CSharpXUnit.Tests/Tests/TestInstrumentation_testTimer.cs
public class TestInstrumentation
{
    [Fact]
    public void TestTimer()
    {
        var timer = new Instrumentation.Timer();

        Assert.Equal(0, timer.GetTicks());
        Assert.Equal(0, timer.GetTotal());
        Assert.Equal(0, timer.GetOwn());
        Assert.Equal(0, timer.GetOwnAvg());
        Assert.Equal(0, timer.GetTotalAvg());
        Assert.Equal(0, timer.GetOwnSquareSum());
        Assert.Equal(0, timer.GetTotalSquareSum());
        Assert.Equal(0, timer.GetOwnMin());
        Assert.Equal(0, timer.GetOwnMax());
        Assert.Equal(0, timer.GetTotalMin());
        Assert.Equal(0, timer.GetTotalMax());

        Assert.Equal(0, timer.GetValue().GetTicks());
        Assert.Equal(0, timer.GetValue().GetTotal());
        Assert.Equal(0, timer.GetValue().GetOwn());
        Assert.Equal(0, timer.GetValue().GetOwnAvg());
        Assert.Equal(0, timer.GetValue().GetTotalAvg());
        Assert.Equal(0, timer.GetValue().GetOwnSquareSum());
        Assert.Equal(0, timer.GetValue().GetTotalSquareSum());
        Assert.Equal(0, timer.GetValue().GetOwnMin());
        Assert.Equal(0, timer.GetValue().GetOwnMax());
        Assert.Equal(0, timer.GetValue().GetTotalMin());
        Assert.Equal(0, timer.GetValue().GetTotalMax());

        var cron1 = new Instrumentation.Cron();
        cron1.Start();
        Thread.Sleep(INTERVAL);
        cron1.Stop();
        timer.AddCron(cron1);

        Assert.Equal(1, timer.GetTicks());
        Assert.Equal(cron1.GetTotal(), timer.GetTotal());
        Assert.Equal(cron1.GetOwn(), timer.GetOwn());
        Assert.Equal(cron1.GetOwn(), timer.GetOwnAvg());
        Assert.Equal(cron1.GetTotal(), timer.GetTotalAvg());
        Assert.Equal(cron1.GetOwn() * cron1.GetOwn(), timer.GetOwnSquareSum());
        Assert.Equal(cron1.GetTotal() * cron1.GetTotal(), timer.GetTotalSquareSum());
        Assert.Equal(cron1.GetOwn(), timer.GetOwnMin());
        Assert.Equal(cron1.GetOwn(), timer.GetOwnMax());
        Assert.Equal(cron1.GetTotal(), timer.GetTotalMin());
        Assert.Equal(cron1.GetTotal(), timer.GetTotalMax());

        Assert.Equal(1, timer.GetValue().GetTicks());
        Assert.Equal(cron1.GetTotal(), timer.GetValue().GetTotal());
        Assert.Equal(cron1.GetOwn(), timer.GetValue().GetOwn());
        Assert.Equal(cron1.GetOwn(), timer.GetValue().GetOwnAvg());
        Assert.Equal(cron1.GetTotal(), timer.GetValue().GetTotalAvg());
        Assert.Equal(cron1.GetOwn() * cron1.GetOwn(), timer.GetValue().GetOwnSquareSum());
        Assert.Equal(cron1.GetTotal() * cron1.GetTotal(), timer.GetValue().GetTotalSquareSum());
        Assert.Equal(cron1.GetOwn(), timer.GetValue().GetOwnMin());
        Assert.Equal(cron1.GetOwn(), timer.GetValue().GetOwnMax());
        Assert.Equal(cron1.GetTotal(), timer.GetValue().GetTotalMin());
        Assert.Equal(cron1.GetTotal(), timer.GetValue().GetTotalMax());

        var cron2 = new Instrumentation.Cron();
        cron2.Start();
        Thread.Sleep(INTERVAL * 2);
        cron2.Stop();
        timer.AddCron(cron2);

        Assert.Equal(2, timer.GetTicks());
        Assert.Equal(cron1.GetTotal() + cron2.GetTotal(), timer.GetTotal());
        Assert.Equal(cron1.GetOwn() + cron2.GetOwn(), timer.GetOwn());
        Assert.Equal((cron1.GetOwn() + cron2.GetOwn()) / 2, timer.GetOwnAvg());
        Assert.Equal((cron1.GetTotal() + cron2.GetTotal()) / 2, timer.GetTotalAvg());
        Assert.Equal(cron1.GetOwn() * cron1.GetOwn() + cron2.GetOwn() * cron2.GetOwn(), timer.GetOwnSquareSum());
        Assert.Equal(cron1.GetTotal() * cron1.GetTotal() + cron2.GetTotal() * cron2.GetTotal(), timer.GetTotalSquareSum());
        Assert.Equal(cron1.GetOwn(), timer.GetOwnMin());
        Assert.Equal(cron2.GetOwn(), timer.GetOwnMax());
        Assert.Equal(cron1.GetTotal(), timer.GetTotalMin());
        Assert.Equal(cron2.GetTotal(), timer.GetTotalMax());
    }
}
